quipucords/api/fact/util.py

```diff
@@ -0,0 +1,183 @@
+#
+# Copyright (c) 2018 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 3 (GPLv3). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv3
+# along with this software; if not, see
+# https://www.gnu.org/licenses/gpl-3.0.txt.
+#
+
+"""Util for validating and persisting source facts."""
+
+import os
+import json
+import copy
+from django.conf import settings
+from django.utils.translation import ugettext as _
+import api.messages as messages
+from api.models import Source, FactCollection
+
+
+ERRORS_KEY = 'errors'
+RESULT_KEY = 'result'
+INVALID_SOURCES_KEY = 'invalid_sources'
+VALID_SOURCES_KEY = 'valid_sources'
+
+# JSON attribute constants
+SOURCES_KEY = 'sources'
+SOURCE_KEY = 'source'
+SOURCE_ID_KEY = 'source_id'
+SOURCE_TYPE_KEY = 'source_type'
+FACTS_KEY = 'facts'
+
+
+def validate_fact_collection_json(fact_collection_json):
+    """Validate fact_collection field.
+
+    :param fact_collection_json: dict representing a fact collection
+    :returns: dict with bool indicating error and result.
+    """
+    if not fact_collection_json.get(SOURCES_KEY):
+        return {ERRORS_KEY: True,
```

Comment:
Couldn't we return a tuple instead of a keyed dictionary?

Reply:
👍